### ISSUE TITLE
 WaveTimelineMaker 

### ISSUE THREAD

**Issue by Avinash01111992** [marytts/marytts#836]
**7. Basic Data Files**
**WaveTimelineMaker**
       The WaveTimelineMaker splits the waveforms as datagrams to be stored in a timeline in Mary format. It produces a binary file, which contains all .wav files.

**BasenameTimelineMaker**
      The BasenameTimelineMaker takes a database root directory and a list of basenames, and associates the basenames with absolute times in a timeline in Mary format.

**MCepTimelineMaker**
     The MCepTimelineMaker takes a database root directory and a list of basenames, and converts the related wav files into a mcep timeline in Mary format.

HI i'm **not able to complete  the above 3 component** in  **Voice Import Components GUI**
Even i'm not able to find whats the error its just showing in red colour please help me out on this ..

**The database.config file is pasted below** 

`# GlobalProperties:
db.allophonesDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/allophones/
db.basenameFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/basenames.lst
db.domain general
db.estDir /usr/
db.gender female
db.halfphoneFeatureDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/halfphonefeatures/
db.halfphoneLabDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/halfphonelab/
db.hplabExtension .hplab
db.labDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/lab/
db.labExtension .lab
db.locale en_US
db.maryServerHost localhost
db.maryServerPort 59125
db.marybase /home/avinash/lb-work/text-to-speach-engine/marytts-5.x/
db.marybaseversion 5.3-SNAPSHOT
db.maryxmlDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/rawmaryxml/
db.maryxmlExtension .xml
db.phoneFeatureDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/phonefeatures/
db.phoneLabDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/phonelab/
db.pmDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/pm/
db.pmExtension .pm
db.promptAllophonesDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/prompt_allophones/
db.ptcDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/ptc/
db.ptcExtension .ptc
db.rootDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/
db.samplingrate 16000
db.textDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/text/
db.textExtension .txt
db.vocalizationsDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/vocalizations/
db.voicename my_voice_avinash
db.wavDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/wav/

# Properties for module PraatPitchmarker:
PraatPitchmarker.command praat --run
PraatPitchmarker.maxPitch 500
PraatPitchmarker.minPitch 100
PraatPitchmarker.pmDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/pm/
PraatPitchmarker.waveDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/wav/

# Properties for module MCEPMaker:
MCEPMaker.mcepDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mcep/

# Properties for module Festvox2MaryTranscripts:
Festvox2MaryTranscripts.transcriptFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/txt.done.data

# Properties for module AllophonesExtractor:
AllophonesExtractor.styleDefinitionFile 

# Properties for module EHMMLabeler:
EHMMLabeler.doAlignment true
EHMMLabeler.doTraining true
EHMMLabeler.eDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/ehmm/
EHMMLabeler.ehmmDir /project/mary/Festival/festvox/src/ehmm/
EHMMLabeler.nonDetEndFlag 0
EHMMLabeler.outputLabDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/lab/
EHMMLabeler.prepareFiles true
EHMMLabeler.promtallophonesDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/prompt_allophones/
EHMMLabeler.reTrainFlag false
EHMMLabeler.startEHMMModelDir /

# Properties for module HTKLabeler:
HTKLabeler.awkbin /usr/bin/awk
HTKLabeler.htDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice//htk/
HTKLabeler.htkDir /usr/bin/
HTKLabeler.maxNoOfIterations 150
HTKLabeler.maxshortPauseIteration 10
HTKLabeler.outputLabDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice//htk/lab/
HTKLabeler.promtallophonesDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice//prompt_allophones/

# Properties for module LabelPauseDeleter:
LabelPauseDeleter.eDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/ehmm/
LabelPauseDeleter.outputLabDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/lab/
LabelPauseDeleter.pauseDurationThreshold 100

# Properties for module HalfPhoneUnitLabelComputer:
HalfPhoneUnitLabelComputer.energyBasedTransientSplitting false

# Properties for module FeatureSelection:
FeatureSelection.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/features.txt
FeatureSelection.maryServerHost localhost
FeatureSelection.maryServerPort 59125

# Properties for module PhoneUnitFeatureComputer:
PhoneUnitFeatureComputer.allophonesDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/allophones/
PhoneUnitFeatureComputer.featureDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/phonefeatures/
PhoneUnitFeatureComputer.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/features.txt
PhoneUnitFeatureComputer.maryServerHost localhost
PhoneUnitFeatureComputer.maryServerPort 59125

# Properties for module HalfPhoneUnitFeatureComputer:
HalfPhoneUnitFeatureComputer.allophonesDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/allophones/
HalfPhoneUnitFeatureComputer.featureDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/halfphonefeatures/
HalfPhoneUnitFeatureComputer.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/features.txt
HalfPhoneUnitFeatureComputer.maryServerHost localhost
HalfPhoneUnitFeatureComputer.maryServerPort 59125

# Properties for module WaveTimelineMaker:
WaveTimelineMaker.waveTimeline /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/timeline_waveforms.mry

# Properties for module BasenameTimelineMaker:
BasenameTimelineMaker.timelineFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/timeline_basenames.mry

# Properties for module MCepTimelineMaker:
MCepTimelineMaker.mcepDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mcep/
MCepTimelineMaker.mcepTimeline /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/timeline_mcep.mry

# Properties for module PhoneUnitfileWriter:
PhoneUnitfileWriter.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/phoneUnits.mry

# Properties for module PhoneFeatureFileWriter:
PhoneFeatureFileWriter.featureDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/phonefeatures/
PhoneFeatureFileWriter.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/phoneFeatures.mry
PhoneFeatureFileWriter.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/phoneUnits.mry
PhoneFeatureFileWriter.weightsFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/phoneUnitFeatureDefinition.txt

# Properties for module DurationCARTTrainer:
DurationCARTTrainer.durTree /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/dur.tree
DurationCARTTrainer.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/phoneFeatures.mry
DurationCARTTrainer.isHnmTimeline false
DurationCARTTrainer.stepwiseTraining false
DurationCARTTrainer.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/phoneUnits.mry
DurationCARTTrainer.waveTimeline /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/timeline_waveforms.mry

# Properties for module F0CARTTrainer:
F0CARTTrainer.f0LeftTreeFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/f0.left.tree
F0CARTTrainer.f0MidTreeFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/f0.mid.tree
F0CARTTrainer.f0RightTreeFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/f0.right.tree
F0CARTTrainer.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/phoneFeatures.mry
F0CARTTrainer.isHnmTimeline false
F0CARTTrainer.stepwiseTraining false
F0CARTTrainer.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/phoneUnits.mry
F0CARTTrainer.waveTimeline /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/timeline_waveforms.mry

# Properties for module HalfPhoneUnitfileWriter:
HalfPhoneUnitfileWriter.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneUnits.mry

# Properties for module HalfPhoneFeatureFileWriter:
HalfPhoneFeatureFileWriter.featureDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/halfphonefeatures/
HalfPhoneFeatureFileWriter.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneFeatures.mry
HalfPhoneFeatureFileWriter.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneUnits.mry
HalfPhoneFeatureFileWriter.weightsFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneUnitFeatureDefinition.txt

# Properties for module F0PolynomialFeatureFileWriter:
F0PolynomialFeatureFileWriter.f0FeatureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/syllableF0Polynomials.mry
F0PolynomialFeatureFileWriter.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneFeatures.mry
F0PolynomialFeatureFileWriter.interpolate true
F0PolynomialFeatureFileWriter.isHnmTimeline false
F0PolynomialFeatureFileWriter.maxPitch 500
F0PolynomialFeatureFileWriter.minPitch 100
F0PolynomialFeatureFileWriter.polynomOrder 3
F0PolynomialFeatureFileWriter.showGraph false
F0PolynomialFeatureFileWriter.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneUnits.mry
F0PolynomialFeatureFileWriter.waveTimeLine /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/timeline_waveforms.mry

# Properties for module AcousticFeatureFileWriter:
AcousticFeatureFileWriter.acFeatDef /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneUnitFeatureDefinition_ac.txt
AcousticFeatureFileWriter.acFeatureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneFeatures_ac.mry
AcousticFeatureFileWriter.contourFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/syllableF0Polynomials.mry
AcousticFeatureFileWriter.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneFeatures.mry
AcousticFeatureFileWriter.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneUnits.mry

# Properties for module JoinCostFileMaker:
JoinCostFileMaker.acfeatureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneFeatures_ac.mry
JoinCostFileMaker.joinCostFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/joinCostFeatures.mry
JoinCostFileMaker.mcepDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mcep/
JoinCostFileMaker.mcepTimeline /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/timeline_mcep.mry
JoinCostFileMaker.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneUnits.mry
JoinCostFileMaker.weightsFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/joinCostWeights.txt

# Properties for module CARTBuilder:
CARTBuilder.acFeatureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneFeatures_ac.mry
CARTBuilder.callWagon false
CARTBuilder.cartFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/cart.mry
CARTBuilder.featureSeqFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/featureSequence.txt
CARTBuilder.maxLeafSize 10000000
CARTBuilder.mcepTimeline /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/timeline_mcep.mry
CARTBuilder.numProcesses 1
CARTBuilder.readFeatureSequence true
CARTBuilder.topLevelTreeFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/topLevel.tree
CARTBuilder.unitFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/halfphoneUnits.mry

# Properties for module HMMVoiceDataPreparation:
HMMVoiceDataPreparation.adaptScripts false
HMMVoiceDataPreparation.userRawDirectory 
HMMVoiceDataPreparation.userUttDirectory 

# Properties for module HMMVoiceConfigure:
HMMVoiceConfigure.adaptF0Ranges 'bdl 40 210 clb 130 260 jmk 50 180 rms 40 200 slt 110 280'
HMMVoiceConfigure.adaptHead b05
HMMVoiceConfigure.adaptScripts false
HMMVoiceConfigure.adaptSpkr slt
HMMVoiceConfigure.adaptSpkrMask */cmu_us_arctic_%%%_*
HMMVoiceConfigure.adaptTrainSpkr 'bdl clb jmk rms'
HMMVoiceConfigure.adaptTransKind feat
HMMVoiceConfigure.adaptTreeKind dec
HMMVoiceConfigure.configureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/hts/configure
HMMVoiceConfigure.dataSet cmu_us_arctic
HMMVoiceConfigure.fftLen 512
HMMVoiceConfigure.frameLen 400
HMMVoiceConfigure.frameShift 80
HMMVoiceConfigure.freqWarp 0.42
HMMVoiceConfigure.gamma 0
HMMVoiceConfigure.lf0BandWidth 1
HMMVoiceConfigure.lnGain 1
HMMVoiceConfigure.lowerF0 110
HMMVoiceConfigure.mgcBandWidth 35
HMMVoiceConfigure.mgcOrder 34
HMMVoiceConfigure.normalize 1
HMMVoiceConfigure.numIterations 5
HMMVoiceConfigure.numState 5
HMMVoiceConfigure.numTestFiles 5
HMMVoiceConfigure.questionsNum 001
HMMVoiceConfigure.sampfreq 16000
HMMVoiceConfigure.speaker slt
HMMVoiceConfigure.strBandWidth 5
HMMVoiceConfigure.strFilterFileName filters/mix_excitation_5filters_99taps_16Kz.txt
HMMVoiceConfigure.strOrder 5
HMMVoiceConfigure.upperF0 280
HMMVoiceConfigure.version 1
HMMVoiceConfigure.windowType 1

# Properties for module HMMVoiceFeatureSelection:
HMMVoiceFeatureSelection.featureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/features.txt
HMMVoiceFeatureSelection.hmmFeatureFile /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/hmmFeatures.txt

# Properties for module HMMVoiceMakeData:
HMMVoiceMakeData.adaptScripts false
HMMVoiceMakeData.featureListFile mary/hmmFeatures.txt
HMMVoiceMakeData.featureListMapFile mary/hmmFeaturesMap.txt
HMMVoiceMakeData.makeCMPMARY 1
HMMVoiceMakeData.makeLABELMARY 1
HMMVoiceMakeData.makeLF0 1
HMMVoiceMakeData.makeLIST 1
HMMVoiceMakeData.makeMGC 1
HMMVoiceMakeData.makeQUESTIONSMARY 1
HMMVoiceMakeData.makeSCP 1
HMMVoiceMakeData.makeSTR 1
HMMVoiceMakeData.questionsFile hts/data/questions/questions_qst001.hed
HMMVoiceMakeData.questionsUttFile hts/data/questions/questions_utt_qst001.hed
HMMVoiceMakeData.trickyPhonesFile mary/trickyPhones.txt

# Properties for module VoiceCompiler:
VoiceCompiler.compileDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/voice-my_voice/
VoiceCompiler.mavenBin /usr/bin/mvn

# Properties for module HMMVoiceCompiler:
HMMVoiceCompiler.compileDir /home/avinash/lb-work/text-to-speach-engine/own-voice-train/myvoice/mary/voice-my_voice-hsmm/
HMMVoiceCompiler.mavenBin /usr/bin/mvn
`

**Comment by psibre** [comment]
TL;DR

What's the error?

**Comment by Avinash01111992** [comment]
This is error i'm getting on running the **WaveTimelineMaker component** 
Processing [0] utterances.

java.lang.ArrayIndexOutOfBoundsException: 0
	at marytts.tools.voiceimport.WaveTimelineMaker.compute(WaveTimelineMaker.java:91)
	at marytts.tools.voiceimport.DatabaseImportMain$8.run(DatabaseImportMain.java:274)
java.lang.ArrayIndexOutOfBoundsException: 0

**Comment by psibre** [comment]
Thanks for the clarification. I'm afraid the message is not terribly user-friendly, but the problem is that you don't have any utterances from which to assemble the waveform timeline. Assuming you did provide a sufficient number of utterances, the cause of the problem is an earlier component, possibly the TranscriptionAligner.

To remedy the situation, I recommend you delete all of the generated data directories *except* for the `wav`, `lab`, and `text` directories, and if you have a `basenames.lst` file, delete that as well. Then re-run each component from the top, one-by-one, keeping an eye on any errors that might occur in the terminal from which you run the `voiceimport` script.

**Comment by Avinash01111992** [comment]
Thank you so much for your response.
Before i start running all the components please clarify that all the components from top to bottom mandatory to run ? or we can skip some of those ?

**Comment by psibre** [comment]
Indeed, the rather lengthy automatic labeling is something that will just waste your time if you run it more than once (unless the predicted phone sequence needs to be changed, of course). In principle, it's better to already have those labels, and skip the EHMMLabeler completely.

**Comment by psibre** [comment]
You'll have to ensure that all of the labels in your `.lab` files are defined in the language's allophone set. In this case, replace the `pau` labels by `_`.

**Comment by psibre** [comment]
And if you find un-alignable transcriptions for a small number of utterances (typically due to errors in pronunciation prediction of proper names, numbers, etc.), see if excluding those utterances helps.

**Comment by Avinash01111992** [comment]
After i replace  pau labels by _    I'm getting the below error 

/home/avinash/lb-work/text-to-speach-engine/TrainMyOwnVoice/ehmm/lab/arctic_a0001.lab
Orig   : _|O|T|r=|_|V|v|_|D|@|_|d|EI|n|dZ|r=|_|t|r|EI|l|_|f|I|l|@|p|_|s|t|i|l|z|_|E|t|s|E|t|r=|@|_
Correct: _|_|O|T|r=|V|v|D|@|d|EI|n|dZ|r=|t|r|EI|l|f|I|l|@|p|s|t|i|l|z|ssil|E|t|s|E|t|r=|@|_
Alignments:  _ _ # O # T # r= # # V # v # # D # @ # # d # EI # n # dZ # r= # # t # r # EI # l # # f # I # l # @ # p # # s # t # i # l # z # # ssil E # t # s # E # t # r= # @ # _ 
Boundary, align =  _ _ 
  deleted boundary from xml
Ph = O, align =  O 
Ph = T, align =  T 
Ph = r=, align =  r= 
Ph = V, align =  V 
Ph = v, align =  v 
Ph = D, align =  D 
Ph = @, align =  @ 
Ph = d, align =  d 
Ph = EI, align =  EI 
Ph = n, align =  n 
Ph = dZ, align =  dZ 
Ph = r=, align =  r= 
Ph = t, align =  t 
Ph = r, align =  r 
Ph = EI, align =  EI 
Ph = l, align =  l 
Boundary, align =  
  deleted boundary from xml
Ph = f, align =  f 
Ph = I, align =  I 
Ph = l, align =  l 
Ph = @, align =  @ 
Ph = p, align =  p 
Ph = s, align =  s 
Ph = t, align =  t 
Ph = i, align =  i 
Ph = l, align =  l 
Ph = z, align =  z 
Boundary, align =  
  deleted boundary from xml
Ph = E, align =  ssil E 
Ph = t, align =  t 
Ph = s, align =  s 
Ph = E, align =  E 
Ph = t, align =  t 
Ph = r=, align =  r= 
Ph = @, align =  @ 
Boundary, align =  _ 
java.lang.Exception: The component TranscriptionAligner produced the following exception: 
	at marytts.tools.voiceimport.DatabaseImportMain$8.run(DatabaseImportMain.java:277)
Caused by: marytts.exceptions.InvalidDataException: Allophone `ssil' could not be found in AllophoneSet `sampa' (Locale: en_US)
	at marytts.tools.analysis.MaryTranscriptionAligner.alignXmlTranscriptions(MaryTranscriptionAligner.java:99)
	at marytts.tools.voiceimport.TranscriptionAligner.alignTranscription(TranscriptionAligner.java:152)
	at marytts.tools.voiceimport.TranscriptionAligner.compute(TranscriptionAligner.java:122)
	at marytts.tools.voiceimport.DatabaseImportMain$8.run(DatabaseImportMain.java:274)

**Comment by Avinash01111992** [comment]
Hi i have completed all the  components but now i struck with the 

Below step 


10. Adding a new unit selection voice in MARY

**VoiceCompiler**

Error is pasted below


Caused by: java.io.IOException: Cannot run program "/usr/bin/mvn" (in directory "/home/avinash/lb-work/text-to-speach-engine/TrainMyOwnVoice/mary/voice-my_voice_demo_avi"): error=2, No such file or directory
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at java.lang.Runtime.exec(Runtime.java:620)
	at java.lang.Runtime.exec(Runtime.java:450)
	at marytts.tools.voiceimport.VoiceCompiler$MavenVoiceCompiler.compileWithMaven(VoiceCompiler.java:368)
	at marytts.tools.voiceimport.VoiceCompiler.compute(VoiceCompiler.java:95)
	at marytts.tools.voiceimport.DatabaseImportMain$8.run(DatabaseImportMain.java:274)
Caused by: java.io.IOException: error=2, No such file or directory
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 5 more

**Comment by Avinash01111992** [comment]
@psibre  Thank you for your help.

Now i'm successfully done with all the steps and now i'm in the publishing my own voice  stage.
**If my new voice is  to be reflected in your marytts server then  master copy of component.xml is must be merged with my local .xml ?**

My question is

**For this to happen i need to be member of marytts   and then once it got merged with master copy then it reflected in GUI ???**

**Comment by psibre** [comment]
Glad to hear you got everything working!

We've actually been building our voices with a newer workflow based on Gradle. We have a dedicated repository for each voice under https://github.com/marytts and the packaged voice is downloadable from each as a release asset. Please have a look at, e.g., https://github.com/marytts/voice-dfki-spike for inspiration.

We're planning to replace the legacy voice publishing documentation very soon -- currently refactoring the website source.

**Comment by Avinash01111992** [comment]
@psibre  
I have done the voice building using  marytts-5.x repository , So now i'm done with my new voice containing .Zip file .
**Now How i can publish  my new voice ? Where i need to add my newvoice.xml component and where i can place my **newVoice.zip**  for providing location link  in xml file ?**

**Comment by psibre** [comment]
Let's start at the bottom.

To load and use your voice, you need a MaryTTS installation (not the source code itself, but what the source code *builds* under `target`). There is a `lib` directory, which is where MaryTTS looks for jar files at startup. File system resources such as the timeline files are placed under `lib/voices`. Install any unit selection voice, and you'll see how this runtime layout works. So to use your voice, you need to unzip the zip file you built into a MaryTTS installation's `lib` directory.

You could also take the zip file you built as well as the component descriptor XML file, and copy them both into a MaryTTS installation's `download` directory. That way, you can run the Component Installer GUI, select your voice, and click "Install".

**Comment by Avinash01111992** [comment]
@psibre  Thanks and clarify  this .
**I need to build the project after i added the zip files and xml files in download and lib directory ?**

**Comment by psibre** [comment]
No before. In fact, you don't even need the source code -- you could also just take a runtime distribution package of MaryTTS or equivalent.

**Comment by Avinash01111992** [comment]
Still i'm not able to see my voice on the GUI (marytts-component-installer)
I have pasted my zip file and xml file in below location of the my mary source 

**/home/avinash/lb-work/text-to-speach-engine/marytts-5.x/target/marytts-5.3-SNAPSHOT/download/** 

**also i have extracted the zip files and placed my new voice into the voices folder(Lib directory) and if i run the marytts-component-installer still i'm not able to find the my voice(i updated the component in GUI as well)**
Where i'm doing wrong please help me out

**Comment by Avinash01111992** [comment]
@psibre  is older workflow is not stable with current version ?

**Comment by Avinash01111992** [comment]
@psibre  Thank yo so much 
According to my understanding i need to wait until **newer Installer, which runs via Gradle**   stable version to  available  ???

**Comment by psibre** [comment]
If you want to open-source your voice data, I recommend you have a look at how we provide our voice repositories on GitHub, and do something similar. Those repositories use a Gradle-based build, *not* the legacy VoiceImportTools GUI.

**Comment by Avinash01111992** [comment]
I am very much fresher  to the git hub and i don't  know how to open source my voice data please help me on this.
**Is i need your permission to add my new voice repository under marytts ??** 
**Please suggest any documentation or me in short ?**

**Comment by psibre** [comment]
1. Read the docs
2. Look at solutions other have found
3. You don't need my permission to publish the MaryTTS voice you built. But we do have certain (undocumented) standards, and our priorities currently lie with completing the core system redesign and associated tooling.

**Comment by Avinash01111992** [comment]
As i noticed in the maryTTs  repository In all the voice open source  
In resources folder the inside voices folder the files that are stored in **.txt format**.
In my case i have generated my voice through legacy voice built tool. my files inside that voices folder is in **.mry format**  .
I'm trying to download the exiting voice open source (Unit selection)  (https://github.com/marytts/voice-dfki-spike) and replacing with my trained new voice files.
Is this the correct way doing to check my voice ??

**Comment by psibre** [comment]
You need to understand that the various directories in the voicebuilding project contain *different types of data*. Please have another look at the old documentation for creating voices (the internals of which are mostly unchanged), and [these](http://www.coli.uni-saarland.de/~steiner/teaching/2013/winter/voicebuilding/intro.pdf) [teaching](http://www.coli.uni-saarland.de/~steiner/teaching/2016/summer/voicebuilding/slides/) [materials](http://www.coli.uni-saarland.de/~steiner/teaching/2016/winter/voicebuilding/slides/).

After that, if you have any specific questions, please open a new issue.
